util/json.c

```diff
@@ -11,6 +12,17 @@ struct json_object *util_json_object_new_double(long double d)
 	if (asprintf(&str, "%Lf", d) < 0)
 		return NULL;
 
+	for (int i = strlen(str) - 1; i > 0; i--)
+	{	// Remove trailing zeros
+		if (str[i] == '.') {
+			str[i] = '\0';
+			break;
+		} else if (str[i] != '0') {
+			str[i+1] = '\0';
```

Comment:
nope
for example
str[] = "1.1"
str[0] == '1', str[1] == '.', str[2] == '1', str[3] == '\0' will be allocated.
so if it's first iteration, it will access [3] which is allocated with null 

https://man7.org/linux/man-pages/man3/asprintf.3.html

The functions asprintf() and vasprintf() are analogs of
       [sprintf(3)](https://man7.org/linux/man-pages/man3/sprintf.3.html) and [vsprintf(3)](https://man7.org/linux/man-pages/man3/vsprintf.3.html), except that they allocate a string
       large enough to hold the output including the terminating null
       byte ('\0'),

Reply:
Ah yes indeed. Thanks for explaining!